app/cloud/server/functions/unregisterWorkspace.js

```diff
@@ -0,0 +1,22 @@
+import { Settings } from '../../../models';
+
+import { retrieveRegistrationStatus } from './retrieveRegistrationStatus';
+
+export function unregisterWorkspace() {
+	const { workspaceRegistered } = retrieveRegistrationStatus();
+	if (!workspaceRegistered) {
+		return true;
+	}
+
+	Settings.updateValueById('Cloud_Workspace_Id', null);
+	Settings.updateValueById('Cloud_Workspace_Name', null);
+	Settings.updateValueById('Cloud_Workspace_Client_Id', null);
+	Settings.updateValueById('Cloud_Workspace_Client_Secret', null);
+	Settings.updateValueById('Cloud_Workspace_Client_Secret_Expires_At', null);
+	Settings.updateValueById('Cloud_Workspace_Registration_Client_Uri', null);
+
+	// So doesn't try to register again automatically
+	Settings.updateValueById('Register_Server', false);
```

Comment:
Don't we inform cloud about the unregistration?

Reply:
No because this isn't actually unregistering.  This just would stop the reporting.  To unregister we are going to force to do via the cloud console that way they are fully authenticated as the account owner and not just a server admin

Reply:
Ah sorry didn't look closely enough at which code segment this was.   This is actually executed after the refresh token is rejected by cloud which causes Rocket.Chat to remove all of these values and effectively is unregistered.

So they unregister on cloud.  The token is no good.  They try to get access token.  Get rejected so then this is executed to clear out all of the stuff and treat as an unregistered server